roles/mirror_registry/tasks/main.yml

```diff
@@ -0,0 +1,55 @@
+---
+# Copyright Red Hat, Inc.
+# All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"); you may
+# not use this file except in compliance with the License. You may obtain
+# a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+# License for the specific language governing permissions and limitations
+# under the License.
+
+- name: Set random auth cred for duration of QE run
+  ansible.builtin.set_fact:
+    # For QE purposes, this only needs to exist for the duration of a single execution.
+    # Randomly generating a string saves us from pulling the randomly generated init password
+    # from the stdout of mirror-registry install.
+    _cifmw_mirror_registry_init_password: "{{ lookup('community.general.random_string', length=12, special=false) }}"
```

Comment:
I think it depends on how the role is used. My initial intention is that this would only be used during a single ci execution. So I wouldn't expect any need to manually interact with it. 

Since it's being set as a fact, I could set it to cacheable which would allow for more broad re-use of the fact in subsequent Ansible execution. But I'm not entirely sure how we would handle the testing of disconnected deployments leveraging this role just yet. My immediate thought is that is should be sufficient to just leave it like this and then reference it when we want to create the pull-secret to be used during the OCP deployment.

Reply:
Yeah I think we should just get this in, it's not used right now so we can't know exactly what's required, the code is good, let's merge it and iterate once the requirements are known.